exporter/opentelemetry-exporter-otlp-proto-grpc/src/opentelemetry/exporter/otlp/proto/grpc/_metric_exporter/__init__.py

```diff
@@ -0,0 +1,123 @@
+    ExportMetricsServiceRequest,
+)
+from opentelemetry.proto.collector.metrics.v1.metrics_service_pb2_grpc import (
+    MetricsServiceStub,
+)
+from opentelemetry.proto.common.v1.common_pb2 import InstrumentationLibrary
+from opentelemetry.proto.metrics.v1.metrics_pb2 import (
+    InstrumentationLibraryMetrics,
+    ResourceMetrics,
+)
+from opentelemetry.proto.metrics.v1.metrics_pb2 import Metric as PB2Metric
+from opentelemetry.sdk._metrics.data import (
+    MetricData,
+)
+
+from opentelemetry.sdk._metrics.export import (
+    MetricExporter,
+    MetricExportResult,
+)
+
+
+class OTLPMetricExporter(
+    MetricExporter,
+    OTLPExporterMixin[MetricData, ExportMetricsServiceRequest, MetricExportResult]
+):
+    _result = MetricExportResult
+    _stub = MetricsServiceStub
+
+    def __init__(
+        self,
+        endpoint: Optional[str] = None,
+        insecure: Optional[bool] = None,
+        credentials: Optional[ChannelCredentials] = None,
+        headers: Optional[Sequence] = None,
+        timeout: Optional[int] = None,
+        compression: Optional[Compression] = None,
+    ):
+        super().__init__(
+            **{
+                "endpoint": endpoint,
+                "insecure": insecure,
+                "credentials": credentials,
+                "headers": headers,
+                "timeout": timeout,
+                "compression": compression,
+            }
+        )
+
+    def _translate_data(
+        self, data: Sequence[MetricData]
+    ) -> ExportMetricsServiceRequest:
+        sdk_resource_instrumentation_library_metrics = {}
+        self._collector_metric_kwargs = {}
+
+        for metric_data in data:
+            resource = metric_data.metric.resource
+            instrumentation_library_map = (
+                sdk_resource_instrumentation_library_metrics.get(resource, {})
+            )
+            if not instrumentation_library_map:
+                sdk_resource_instrumentation_library_metrics[
+                    resource
+                ] = instrumentation_library_map
+
+            instrumentation_library_metrics = instrumentation_library_map.get(
+                metric_data.instrumentation_info
+            )
+
+            if not instrumentation_library_metrics:
+                if metric_data.instrumentation_info is not None:
+                    instrumentation_library_map[
+                        metric_data.instrumentation_info
+                    ] = InstrumentationLibraryMetrics(
+                        instrumentation_library=InstrumentationLibrary(
+                            name=metric_data.instrumentation_info.name,
+                            version=metric_data.instrumentation_info.version,
+                        )
+                    )
+                else:
+                    instrumentation_library_map[
+                        metric_data.instrumentation_info
+                    ] = InstrumentationLibraryMetrics()
```

Comment:
Can the constructor be called like this? Is there a default value for `instrumentation_library` if called with no params?

Reply:
It appears so...
https://github.com/open-telemetry/opentelemetry-python/blob/d8bebdd5f7c3f08dd565a71be5d84c5ab500c32b/exporter/opentelemetry-exporter-otlp-proto-grpc/src/opentelemetry/exporter/otlp/proto/grpc/trace_exporter/__init__.py#L271
https://github.com/open-telemetry/opentelemetry-python/blob/d8bebdd5f7c3f08dd565a71be5d84c5ab500c32b/exporter/opentelemetry-exporter-otlp-proto-grpc/src/opentelemetry/exporter/otlp/proto/grpc/_log_exporter/__init__.py#L147

Reply:
Perhaps this case just never has been hit (instrumentation info has always been present in the incoming spans/logs)? I might be misreading the proto file https://github.com/open-telemetry/opentelemetry-python/blob/39fe4dbc7a5b7f125fa9988ecdb7b26dc638cc9f/opentelemetry-proto/src/opentelemetry/proto/metrics/v1/metrics_pb2.py#L123

Reply:
Created an issue to track this since it will need to be fixed for all signals if this is a problem https://github.com/open-telemetry/opentelemetry-python/issues/2356